In [80]:
import pandas as pd
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/googleplaystore.csv"
df = pd.read_csv(url)

----

Pregunta A

In [81]:
duplicados = df.duplicated(subset=['App'])
print("Número de filas duplicadas: ", duplicados.sum())

Número de filas duplicadas:  1181


In [82]:
df.drop_duplicates(subset=['App'], inplace=True)


¿Cuántos registros duplicados purgaron?

Se encontrar y eliminaron 1181 registros duplicados.

----

Pregunta B

In [83]:
df[df['Installs'] == 'Free']

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
10472,Life Made WI-Fi Touchscreen Photo Frame,1.9,19.0,3.0M,"1,000+",Free,0,Everyone,NaN,"February 11, 2018",1.0.19,4.0 and up,NaN


In [84]:
df = df[df['Installs'].ne('Free')].copy()

df['Installs'] = (
    df['Installs']
    .astype(str)
    .str.replace('+', '', regex=False)
    .str.replace(',', '', regex=False)
    .astype(int)
)

In [85]:
df['Installs'].mean()

np.float64(7777506.732270421)

¿Cuál es el promedio real de descargas?

El promedio de instalacion es aprox 7.7 millones de descargas por aplicacion.

----

Pregunta C

In [86]:
df['Price'] = (
    df['Price']
    .str.replace('$', '', regex=False)
    .astype(float)
)

In [87]:
df['Price'].dtype

dtype('float64')

In [88]:
df[['App', 'Price']].head()

,App,Price
0,Photo Editor & Candy Camera & Grid & ScrapBook,0.0
1,Coloring book moana,0.0
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",0.0
3,Sketch - Draw & Paint,0.0
4,Pixel Draw - Number Art Coloring Book,0.0


----

Pregunta D

In [89]:
df['Price'].max()

np.float64(400.0)

In [90]:
df[df['Price'] > 200]

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
4197,most expensive app (H),FAMILY,4.3,6,1.5M,100,Paid,399.99,Everyone,Entertainment,"July 16, 2018",1.0,7.0 and up
4362,💎 I'm rich,LIFESTYLE,3.8,718,26M,10000,Paid,399.99,Everyone,Lifestyle,"March 11, 2018",1.0.0,4.4 and up
4367,I'm Rich - Trump Edition,LIFESTYLE,3.6,275,7.3M,10000,Paid,400.00,Everyone,Lifestyle,"May 3, 2018",1.0.1,4.1 and up
5351,I am rich,LIFESTYLE,3.8,3547,1.8M,100000,Paid,399.99,Everyone,Lifestyle,"January 12, 2018",2.0,4.0.3 and up
5354,I am Rich Plus,FAMILY,4.0,856,8.7M,10000,Paid,399.99,Everyone,Entertainment,"May 19, 2018",3.0,4.4 and up
5355,I am rich VIP,LIFESTYLE,3.8,411,2.6M,10000,Paid,299.99,Everyone,Lifestyle,"July 21, 2018",1.1.1,4.3 and up
5356,I Am Rich Premium,FINANCE,4.1,1867,4.7M,50000,Paid,399.99,Everyone,Finance,"November 12, 2017",1.6,4.0 and up
5357,I am extremely Rich,LIFESTYLE,2.9,41,2.9M,1000,Paid,379.99,Everyone,Lifestyle,"July 1, 2018",1.0,4.0 and up
5358,I am Rich!,FINANCE,3.8,93,22M,1000,Paid,399.99,Everyone,Finance,"December 11, 2017",1.0,4.1 and up
5359,I am rich(premium),FINANCE,3.5,472,965k,5000,Paid,399.99,Everyone,Finance,"May 1, 2017",3.4,4.4 and up


In [ ]:
df = df[df['Price'] < 50]

In [107]:
df.to_csv('playstore_limpio.csv', index=False)